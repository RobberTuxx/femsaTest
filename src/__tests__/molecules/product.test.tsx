import React from 'react';
import renderer from 'react-test-renderer';
import Product from "../../components/molecules/product";
const mockedDispatch = jest.fn();

// Mocks like this need to be configured at the top level
// of the test file, they can't be setup inside your tests.
jest.mock("@react-navigation/native", () => {
    const actualNav = jest.requireActual("@react-navigation/native");
    return {
        ...actualNav,
        useNavigation: () => ({
            navigate: jest.fn(),
            dispatch: mockedDispatch,
        }),
    };
});

describe('test nav', ()=>{
    beforeEach(() => {
        // Alternatively, set "clearMocks" in your Jest config to "true"
        mockedDispatch.mockClear();
    });
    it('render Product', () => {
        const component = renderer.create(<Product product={{
            id: "1",
            points: 1000,
            product: 'product',
            is_redemption: true,
            createdAt: "2023-03-11T12:17:47.118Z",
            image: "https://loremflickr.com/640/480/business"
        }}/>).toJSON()
        expect(component).toMatchSnapshot()
    })
    it('has 0 child', () => {
        const component = renderer.create(<Product product={{
            id: "1",
            points: 1000,
            product: 'product',
            is_redemption: true,
            createdAt: "2023-03-11T12:17:47.118Z",
            image: "https://loremflickr.com/640/480/business"
        }}/>).toJSON()
        expect(component.children.length).toBe(5)
    })
})


