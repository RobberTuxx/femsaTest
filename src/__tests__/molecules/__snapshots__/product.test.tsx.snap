// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Product 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "flexDirection": "row",
      "marginVertical": 12.5,
      "opacity": 1,
      "width": 630,
    }
  }
>
  <Image
    source={
      {
        "uri": "https://loremflickr.com/640/480/business",
      }
    }
    style={
      {
        "borderRadius": 10,
        "height": 86,
        "width": 86,
      }
    }
  />
  <View
    style={
      {
        "marginLeft": 19,
        "width": 362.5,
      }
    }
  >
    <Text
      style={
        {
          "color": "#000000",
          "fontFamily": "Avenir",
          "fontSize": 22,
          "fontWeight": "800",
        }
      }
    >
      product
    </Text>
    <Text
      style={
        {
          "color": "#000000",
          "fontFamily": "Avenir",
          "fontSize": 19,
          "fontWeight": "400",
        }
      }
    >
      11 de marzo de 2023
    </Text>
  </View>
  <Text
    style={
      {
        "color": "#FF0000",
        "fontSize": 25,
      }
    }
  >
    -
  </Text>
  <Text
    style={
      {
        "color": "#000000",
        "fontFamily": "Avenir",
        "fontSize": 25,
        "fontWeight": "800",
        "width": 124,
      }
    }
  >
     
    1000
  </Text>
  <Text
    style={
      {
        "fontWeight": "800",
      }
    }
  >
    &gt;
  </Text>
</View>
`;

exports[`test nav render Product 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "flexDirection": "row",
      "marginVertical": 12.5,
      "opacity": 1,
      "width": 630,
    }
  }
>
  <Image
    source={
      {
        "uri": "https://loremflickr.com/640/480/business",
      }
    }
    style={
      {
        "borderRadius": 10,
        "height": 86,
        "width": 86,
      }
    }
  />
  <View
    style={
      {
        "marginLeft": 19,
        "width": 362.5,
      }
    }
  >
    <Text
      style={
        {
          "color": "#000000",
          "fontFamily": "Avenir",
          "fontSize": 22,
          "fontWeight": "800",
        }
      }
    >
      product
    </Text>
    <Text
      style={
        {
          "color": "#000000",
          "fontFamily": "Avenir",
          "fontSize": 19,
          "fontWeight": "400",
        }
      }
    >
      11 de marzo de 2023
    </Text>
  </View>
  <Text
    style={
      {
        "color": "#FF0000",
        "fontSize": 25,
      }
    }
  >
    -
  </Text>
  <Text
    style={
      {
        "color": "#000000",
        "fontFamily": "Avenir",
        "fontSize": 25,
        "fontWeight": "800",
        "width": 124,
      }
    }
  >
     
    1000
  </Text>
  <Text
    style={
      {
        "fontWeight": "800",
      }
    }
  >
    &gt;
  </Text>
</View>
`;
